src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,397 @@
+  const std::filesystem::path& cmd,
+  ss::sstring& filename,
+  const debug_bundle_params& params,
+  std::optional<debug_bundle_credentials>& creds) {
+    std::vector<ss::sstring> rpk_argv{
+      cmd.string(), "debug", "bundle", "--output", filename};
+    // Add SASL creds to RPK flags if SASL is enabled.
+    // No need to check for sasl on the broker endpoint because that is for
+    // Kafka brokers where as the bundle is managed by the Admin server only.
+    // TODO(@Nyalia): Check if `rpk debug bundle` issues Kafka requests. If so,
+    // then we also need to check for sasl on the kafka brokers.
+    if (creds.has_value() && config::shard_local_cfg().enable_sasl()) {
+        rpk_argv.push_back("--user");
+        rpk_argv.push_back(creds->username);
+        rpk_argv.push_back("--password");
+        rpk_argv.push_back(creds->password);
+        rpk_argv.push_back("--sasl-mechanism");
+        rpk_argv.push_back(creds->mechanism);
+    }
+
+    if (params.logs_since.has_value()) {
+        rpk_argv.push_back("--logs-since");
+        rpk_argv.push_back(params.logs_since.value());
+    }
+
+    if (params.logs_until.has_value()) {
+        rpk_argv.push_back("--logs-until");
+        rpk_argv.push_back(params.logs_until.value());
+    }
+
+    if (params.logs_size_limit.has_value()) {
+        rpk_argv.push_back("--logs-size-limit");
+        rpk_argv.push_back(params.logs_size_limit.value());
+    }
+
+    if (params.metrics_interval.has_value()) {
+        rpk_argv.push_back("--metrics-interval");
+        rpk_argv.push_back(params.metrics_interval.value());
+    }
+
+    return rpk_argv;
+}
+
+std::error_code detail::handle_wait_status(
+  ss::experimental::process::wait_status& wstatus,
+  ss::sstring cmd,
+  ss::sstring filename) {
+    auto* exit_status = std::get_if<ss::experimental::process::wait_exited>(
+      &wstatus);
+    if (exit_status != nullptr) {
+        if (exit_status->exit_code != 0) {
+            vlog(
+              bundle_log.error,
+              "Failed to run {}, exit code {}",
+              cmd,
+              exit_status->exit_code);
+        } else {
+            vlog(
+              bundle_log.debug,
+              "{} successful, debug bundle {}",
+              cmd,
+              filename);
+            return rpk::debug_bundle_error::success;
+        }
+    } else {
+        auto* exit_signal
+          = std::get_if<ss::experimental::process::wait_signaled>(&wstatus);
+        if (exit_signal == nullptr) {
+            vlog(
+              bundle_log.error,
+              "Failed to run {} and the exit "
+              "signal is undefined. Debug bundle "
+              "{}",
+              cmd,
+              filename);
+        } else {
+            vlog(
+              bundle_log.error,
+              "Failed to run {}, process "
+              "terminated with signal {}",
+              cmd,
+              exit_signal->terminating_signal);
+        }
+    }
```

Comment:
nit: This may be simpler if you use `std::visit`: https://en.cppreference.com/w/cpp/utility/variant/visit

```c++

template<class>
inline constexpr bool always_false_v = false;

std::visit([](auto && arg) {
  using T = std::decay_t<decltype(arg)>;
  if constexpr (std::is_same_v<T, ss::experiemental::process::wait_exited>) {
    arg.exit_code;
  } else if constexpr(std::is_same_v<T, ss::experimental::process::wait_signaled>) {
    arg.terminating_signal;
  } else {
    static_assert(always_false_v<T>, "non-exhaustive search");
  }
}, wstatus);
```

Reply:
Personally, I do not think using `std::visit` makes the code any more simpler than it already is. Currently, it is just a couple if-statements. I've also removed the `cmd` and `filename` args from the method which you will see in the next push.

Reply:
The reason that you may want to consider using a visit is because it is type safe. You can use the seastar helper like this:

```
ss::visit(
  [](ss::experiemental::process::wait_exited) { ... },
  [](ss::experiemental::process::wait_signaled) { ... },
  wstatus);
```

By type safe what I mean is that the code will not compile unless you cover all the cases. If seastar adds a new type to the wstatus std::variant, then you'll be guaranteed to know.